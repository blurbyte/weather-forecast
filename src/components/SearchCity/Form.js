// Form hidden on tablets or smaller devices

import styled from 'styled-components';

import media from '../../styles/media';

const Form = styled.form`
  position: relative;

  ${media.tablet`
    display: none;
  `};
`;

export default Form;
