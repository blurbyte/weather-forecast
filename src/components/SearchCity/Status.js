// Small dropdown below input

import styled from 'styled-components';
import List from './List';

const Base = List.withComponent('div');

const Status = styled(Base)`
  display: flex;
  align-items: center;
  height: 4rem;
  padding: 0 1.4rem;
  font-style: italic;
  color: rgba(34, 34, 34, 0.5);
`;

export default Status;
