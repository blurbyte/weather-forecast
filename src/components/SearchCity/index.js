// City search input and result dropdown

import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Search from '../../containers/Search';
import SearchInput from '../SearchInput';
import SearchCityResult from '../SearchCityResult';
import Form from './Form';
import Status from './Status';
import List from './List';
import ListItem from './ListItem';

class SearchCity extends Component {
  static propTypes = {
    fetchForecast: PropTypes.func.isRequired
  };

  render() {
    return (
      <Search fetchForecast={this.props.fetchForecast}>
        {({
          search,
          foundCities,
          nothingFound,
          handleSearch,
          handleSubmit,
          fetchForecast,
          resetFoundCities,
          resetSearch
        }) => (
          <Form onSubmit={handleSubmit}>
            <SearchInput
              placeholder="Your city name"
              value={search}
              onChange={handleSearch}
              onBlur={resetFoundCities}
            />
            {foundCities.length > 0 && (
              <SearchCityResult list={List} listItem={ListItem} cities={foundCities} fetchForecast={fetchForecast} />
            )}
            {!foundCities.length && nothingFound && <Status onMouseDown={resetSearch}>No cities found</Status>}
          </Form>
        )}
      </Search>
    );
  }
}

export default SearchCity;
