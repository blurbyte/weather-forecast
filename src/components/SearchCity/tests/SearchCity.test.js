import React from 'react';
import renderer from 'react-test-renderer';

import SearchCity from '../index';

jest.mock('../../SearchInput', () => 'SearchInput');
jest.mock('../../Icons', () => ({
  Magnifier: 'Magnifier'
}));

test('renders correctly', () => {
  const component = renderer.create(<SearchCity fetchForecast={jest.fn()} />);
  expect(component.toJSON()).toMatchSnapshot();
});
