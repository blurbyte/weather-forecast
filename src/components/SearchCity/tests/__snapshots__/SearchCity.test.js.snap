// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="sc-htpNat drYivi"
  onSubmit={[Function]}
>
  <SearchInput
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Your city name"
    value=""
  />
</form>
`;
